app/controllers/api_controller/parameters.rb

```diff
@@ -28,26 +28,38 @@ def hash_fetch(hash, element, default = {})
     def filter_param(klass)
       return nil if params['filter'].blank?
 
-      operators = {"!=" => {:default => :not_eq, :str => :does_not_match},
+      operators = {"!=" => {:default => :not_eq, :str => :does_not_match, :ruby => :!=},
                    "<=" => {:default => :lteq},
                    ">=" => {:default => :gteq},
                    "<"  => {:default => :lt},
                    ">"  => {:default => :gt},
-                   "="  => {:default => :eq, :str => :matches}
+                   "="  => {:default => :eq, :str => :matches, :ruby => :==}
                   }
 
       res_filter = nil
+      ruby_filters = []
+
       params['filter'].select(&:present?).each do |filter|
         parsed_filter = parse_filter(filter, operators)
-
-        arel = klass.arel_table[parsed_filter[:attr]].send(parsed_filter[:method], parsed_filter[:value])
-        res_filter = if res_filter.nil?
-                       arel
-                     else
-                       parsed_filter[:logical_or] ? res_filter.or(arel) : res_filter.and(arel)
-                     end
+        if klass.column_names.include?(parsed_filter[:attr].to_s)
+          arel = klass.arel_table[parsed_filter[:attr]].send(parsed_filter[:method], parsed_filter[:value])
+          res_filter = if res_filter.nil?
+                         arel
+                       else
+                         parsed_filter[:logical_or] ? res_filter.or(arel) : res_filter.and(arel)
+                       end
+        elsif klass.virtual_column?(parsed_filter[:attr])
+          ruby_filters << lambda do |result_set|
+            result_set.select do |resource|
+              attr = resource.public_send("#{parsed_filter[:attr]}")
+              attr.send("#{parsed_filter[:ruby_operator]}", parsed_filter[:value])
+            end
+          end
+        else
+          raise BadRequestError, "attribute #{parsed_filter[:attr]} does not exist"
```

Comment:
:+1: catch them early before we get 500's with PG::UndefinedColumn error.  Thanks.